cairo/ethereum/cancun/state.cairo

```diff
@@ -181,3 +190,109 @@ func get_storage{poseidon_ptr: PoseidonBuiltin*, state: State}(
 
     return value;
 }
+
+func set_storage{poseidon_ptr: PoseidonBuiltin*, state: State}(
+    address: Address, key: Bytes32, value: U256
+) {
+    alloc_locals;
+
+    let storage_tries = state.value._storage_tries;
+    let fp_and_pc = get_fp_and_pc();
+    local __fp__: felt* = fp_and_pc.fp_val;
+
+    // Assert that the account exists
+    let account = get_account_optional(address);
+    if (cast(account.value, felt) == 0) {
+        // TODO: think about which cases lead to this error and decide on the correct type of exception to raise
+        // perhaps AssertionError
+        with_attr error_message("Cannot set storage on non-existent account") {
+            assert 0 = 1;
+        }
+    }
+
+    let storage_tries_dict_ptr = cast(storage_tries.value.dict_ptr, DictAccess*);
+    // Use `hashdict_get` instead of `hashdict_read` because `MappingAddressTrieBytes32U256` is not a
+    // `default_dict`. Accessing a key that does not exist in the dict would have panicked for `hashdict_read`.
+    let (storage_trie_pointer) = hashdict_get{
+        poseidon_ptr=poseidon_ptr, dict_ptr=storage_tries_dict_ptr
+    }(1, &address.value);
+
+    if (cast(storage_trie_pointer, felt) == 0) {
+        let (new_mapping_dict_ptr) = dict_new();
```

Comment:
Does not it needs a %{initial_dict = {} %} ?
https://github.com/starkware-libs/cairo-lang/blob/8e11b8cc65ae1d0959328b1b4a40b92df8b58595/src/starkware/cairo/common/dict.cairo#L7